internal/events/event_stream_producer.go

```diff
@@ -0,0 +1,101 @@
+package events
+
+import (
+	c "github.com/RedHatInsights/sources-api-go/config"
+	"github.com/RedHatInsights/sources-api-go/dao"
+	"github.com/RedHatInsights/sources-api-go/kafka"
+	logging "github.com/RedHatInsights/sources-api-go/logger"
+	m "github.com/RedHatInsights/sources-api-go/model"
+	"github.com/RedHatInsights/sources-api-go/util"
+)
+
+const EventStreamTopic = "platform.sources.event-stream"
+
+var config = c.Get()
+
+type EventStreamProducer struct {
+	Sender
+}
+
+type Sender interface {
+	RaiseEvent(eventType string, payload []byte, headers []kafka.Header) error
+}
+
+type EventStreamSender struct {
+}
+
+func (esp *EventStreamSender) RaiseEvent(eventType string, payload []byte, headers []kafka.Header) error {
+	logging.Log.Debugf("publishing message to topic %q...", EventStreamTopic)
+
+	producerConfig := kafka.ProducerConfig{Topic: config.KafkaTopic(EventStreamTopic)}
+	kafkaConfig := kafka.Config{KafkaBrokers: config.KafkaBrokers, ProducerConfig: producerConfig}
+	kf := &kafka.Manager{Config: kafkaConfig}
+
+	m := &kafka.Message{}
+
+	for index, header := range headers {
```

Comment:
Can probably switch this loop out for `m.GetHeader("event_type")`

Reply:
Here I am no sure  - how do you mean it ?

`m.GetHeader("event_type")` finds and returns in m.Headers value with "event_type" key.

but what I need here is to replace value with key "event_type"  in `headers` as header with key "event_type" is already present in `headers` because those headers came from message sent to kafka with status message - header with "event_type" is required here(./statuslistener/statuslistener.go:64)